304-known-implementations.md

```diff
@@ -0,0 +1,123 @@
+
+#### Using Signy to perform TUF verifications
+
+- signing and pushing a bundle:
+
+```
+$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 sign testdata/cnab/bundle.json localhost:5000/thin-bundle:v1
+INFO[0000] Pushed trust data for localhost:5000/thin-bundle:v1: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+INFO[0000] Starting to copy image cnab/helloworld:0.1.1
+INFO[0002] Completed image cnab/helloworld:0.1.1 copy
+INFO[0002] Generated relocation map: relocation.ImageRelocationMap{"cnab/helloworld:0.1.1":"localhost:5000/thin-bundle@sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6"}
+INFO[0002] Pushed successfully, with digest "sha256:b4936e42304c184bafc9b06dde9ea1f979129e09a021a8f40abc07f736de9268"
+```
+
+- pulling from an OCI registry and verifying the signature of a bundle:
+
+```
+$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 verify localhost:5000/thin-bundle:v1
+INFO[0000] Pulled trust data for localhost:5000/thin-bundle:v1, with role targets - SHA256: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+INFO[0000] Pulling bundle from registry: localhost:5000/thin-bundle:v1
+INFO[0000] Computed SHA: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+INFO[0000] The SHA sums are equal: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+```
+
+#### Using Signy to also perform in-toto verifications
+
+> Note that currently Signy does _not_ properly handle the signing key of the in-toto root layout file, which should either be distributed using TUF, or out of band.
+
+- when signing a bundle, if in-toto metadata is available, it can be added to the `custom` object of the TUF targets file:
+
+```
+$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 sign testdata/cnab/bundle.json localhost:5000/thin-intoto:v2 --in-toto --layout testdata/intoto/demo.layout.template --links testdata/intoto --layout-key testdata/intoto/alice.pub
+INFO[0000] Adding In-Toto layout and links metadata to TUF
+INFO[0000] Pushed trust data for localhost:5000/thin-intoto:v2: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
+INFO[0000] Starting to copy image cnab/helloworld:0.1.1
+INFO[0001] Completed image cnab/helloworld:0.1.1 copy
+INFO[0001] Generated relocation map: relocation.ImageRelocationMap{"cnab/helloworld:0.1.1":"localhost:5000/thin-intoto@sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6"}
+INFO[0001] Pushed successfully, with digest "sha256:b4936e42304c184bafc9b06dde9ea1f979129e09a021a8f40abc07f736de9268"
+```
+
+- if the TUF metadata associated with a bundle also contains in-toto metadata in the `custom` object of the targets file, Signy will validate all layouts and links, and perform the verifications inside a verification image:
```

Comment:
[Ditto](https://github.com/deislabs/cnab-spec/pull/288/files#r337770475)

Reply:
This is still an open work item for Signy.